base/test.jl

```diff
@@ -908,4 +909,16 @@ function detect_ambiguities(mods...; imported::Bool=false)
     collect(ambs)
 end
 
+"""
+The `GenericString` can be used to test generic string APIs that program to
+the `AbstractString` interface. In order to ensure that functions can work
+with string types besides the standard `String` type.
+"""
+immutable GenericString <: AbstractString
```

Comment:
TestHelpers currently only gets used by base's tests, but it is basically serving this same purpose of placeholder types that only get used for mocking. The actual test framework is for defining and running them, not providing data to run them on.